src/Tools/TestDiscoveryWorker/Program.cs

```diff
@@ -12,96 +12,102 @@
-    testDescriptor += " (.NET Framework)";
+    string tfm = "(.NET Framework)";
 #endif
 
-        await Console.Out.WriteLineAsync($"Discovering tests in {testDescriptor}...").ConfigureAwait(false);
+    Console.Write($"Discovering tests in {tfm} {assemblyFileName} ... ");
 
-        using var xunit = new XunitFrontController(AppDomainSupport.IfAvailable, assemblyFileName, shadowCopy: false);
-        var configuration = ConfigReader.Load(assemblyFileName, configFileName: null);
-        var sink = new Sink();
-        xunit.Find(includeSourceInformation: false,
-                   messageSink: sink,
-                   discoveryOptions: TestFrameworkOptions.ForDiscovery(configuration));
+    using var xunit = new XunitFrontController(AppDomainSupport.IfAvailable, assemblyFilePath, shadowCopy: false);
+    var configuration = ConfigReader.Load(assemblyFileName, configFileName: null);
+    var sink = new Sink();
+    xunit.Find(includeSourceInformation: false,
+                messageSink: sink,
+                discoveryOptions: TestFrameworkOptions.ForDiscovery(configuration));
 
-        var testsToWrite = new HashSet<string>();
-        await foreach (var fullyQualifiedName in sink.GetTestCaseNamesAsync())
-        {
-            testsToWrite.Add(fullyQualifiedName);
-        }
-
-        if (sink.AnyWriteFailures)
-        {
-            await Console.Error.WriteLineAsync($"Channel failed to write for '{assemblyFileName}'").ConfigureAwait(false);
-            return ExitFailure;
-        }
-
-#if NET6_0_OR_GREATER
-        await Console.Out.WriteLineAsync($"Discovered {testsToWrite.Count} tests in {testDescriptor}").ConfigureAwait(false);
-#else
-        await Console.Out.WriteLineAsync($"Discovered {testsToWrite.Count} tests in {testDescriptor}").ConfigureAwait(false);
-#endif
+    var testsToWrite = new HashSet<string>();
+    await foreach (var fullyQualifiedName in sink.GetTestCaseNamesAsync())
+    {
+        testsToWrite.Add(fullyQualifiedName);
+    }
 
-        var directory = Path.GetDirectoryName(assemblyFileName);
-        using var fileStream = File.Create(Path.Combine(directory!, "testlist.json"));
-        await JsonSerializer.SerializeAsync(fileStream, testsToWrite).ConfigureAwait(false);
-        return ExitSuccess;
+    if (sink.AnyWriteFailures)
+    {
+        Console.WriteLine($"Channel failed to write for '{assemblyFileName}'");
+        return ExitFailure;
     }
 
+    Console.WriteLine($"{testsToWrite.Count} found");
+
+    using var fileStream = new FileStream(outputFilePath, FileMode.OpenOrCreate, FileAccess.Write, FileShare.None);
+    await JsonSerializer.SerializeAsync(fileStream, testsToWrite.OrderBy(x => x)).ConfigureAwait(false);
+    return ExitSuccess;
+}
+catch (OptionException e)
+{
+    Console.WriteLine(e.Message);
+    options.WriteOptionDescriptions(Console.Out);
     return ExitFailure;
 }
 catch (Exception ex)
 {
     // Write the exception details to stderr so the host process can pick it up.
```

Comment:
It doesn't look like we write to stderr anymore, i.e., the comment is incorrect.

Reply:
Good catch. Will fix this on the next change.